dashboard/app/assets/javascripts/header.js.erb

```diff
@@ -413,3 +416,47 @@ dashboard.header.updateTimestamp = function () {
     $('.project_updated_at').text("Not saved"); // TODO i18n
   }
 };
+
+function getSummarizedProgressForAnonymousUser () {
+  var summarizedProgress = {};
+  var levelProgress = {};
+  var cookie = document.cookie.split(';');
+  
+  for(var i = 0; i < cookie.length; i++) {
+    var pair = cookie[i].split('=');
+    var key = pair[0].trim();
+
+    switch(key) {
+      case 'progress':
+        var allLevelsProgress = JSON.parse(decodeURIComponent(pair[1]));
+        for(var level in allLevelsProgress) {
+          if (allLevelsProgress.hasOwnProperty(level)) {
+            var individualLevelProgressDescription = '';
+            var individualLevelProgress = allLevelsProgress[level];
+
+            //TODO: Find a way to use the constants defined somewhere
```

Comment:
There are ruby constants for these values so you can assign their values in an erb file; 

e.g  var x = #{Foo.CLASS_CONSTANT}

I would suggest keeping as much of the code as possible free of Ruby interpolation so that we can split out the truly static scripts from those that require Ruby execution.  (This will help with lintng and efficiency.)